lib/galaxy/tool_util/linters/inputs.py

```diff
@@ -276,6 +355,77 @@ def lint_inputs(tool_xml, lint_ctx):
+    for param in inputs:
+        try:
+            param_name = _parse_name(param.attrib.get("name"), param.attrib.get("argument"))
+        except ValueError:
+            continue
+        if _param_in_compiled_cheetah(param, param_name, code):
+            continue
+        if param_name in filter_code:
+            continue
+        if tool_xml.find("./inputs//param/options[@from_dataset]") is not None or tool_xml.find("./inputs//param/options/filter[@ref]") is not None:
+            continue
+        if param.getparent().tag == "conditional":
+            continue
+
+        conf_inputs = tool_xml.find("./configfiles/inputs")
+        if conf_inputs is not None:  # in this it's really hard to know
+            param_type = param.attrib.get("type")
+            if param_type in ["data", "collection"]:
+                if not conf_inputs.attrib.get("data_style"):
+                    lint_ctx.error(f"Param input [{param_name}] not found in command or configfiles. Does the present inputs config miss the 'data_style' attribute?", line=param.sourceline, xpath=tool_xml.getpath(param))
+            inputs_conf_name = conf_inputs.attrib.get('name', conf_inputs.attrib.get('filename', None))
+            if inputs_conf_name:
+                if not re.search(r"(^|[^\w])" + inputs_conf_name + r"([^\w]|$)", code):
+                    lint_ctx.error(f"Param input [{param_name}] only used inputs configfile {inputs_conf_name}, but this is not used in the command", line=param.sourceline, xpath=tool_xml.getpath(param))
+            continue
+
+        change_format = tool_xml.find(f"./outputs//change_format/when[@input='{param_name}']") is not None
+        template = re.search(r"(^|[^\w])" + param_name + r"([^\w]|$)", template_code) is not None
+        action = re.search(r"(^|[^\w])" + param_name + r"([^\w]|$)", action_code) is not None
+        label = re.search(r"[^\w]" + param_name + r"([^\w]|$)", label_code) is not None
+        if template or action or label:
+            if template + action + label == 1:
+                only = "only "
+            else:
+                only = ""
+            # TODO check that template is used??
+            if template:
+                lint_ctx.warn(f"Param input [{param_name}] {only}used in a template macro, use a macro token instead.", line=param.sourceline, xpath=tool_xml.getpath(param))
+            if action:
+                lint_ctx.warn(f"Param input [{param_name}] {only}found in output actions, better use extended metadata.", line=param.sourceline, xpath=tool_xml.getpath(param))
+            if label:
+                lint_ctx.warn(f"Param input [{param_name}] {only}found in a label attribute, this is discouraged.", line=param.sourceline, xpath=tool_xml.getpath(param))
+            continue
+
+        if change_format:
+            lint_ctx.error(f"Param input [{param_name}] is only used in a change_format tag", line=param.sourceline, xpath=tool_xml.getpath(param))
```

Comment:
Why would this be an error ?

Reply:
In my understanding the `change_format` tag tells the tool: depending on the parameter `param_name` the generated command line creates output of a different format .. which is then set for the output dataset.

I can't think of a scenario in which the generated command line would create output of a different format without using the parameter `param_name` at all.

Reply:
OK. Will make this a warning. Found a use case https://github.com/ARTbio/tools-artbio/blob/667afd5a5408321a31cdebab75b378e694deab07/tools/justgzip/gzip.xml#L20 .. even if this might be better over there: https://github.com/ARTbio/tools-artbio/pull/580